javalib/src/main/scala/java/util/Formatter.scala

```diff
@@ -945,6 +955,27 @@ object Formatter {
   private val FormatSpecifier = new js.RegExp(
       """(?:(\d+)\$)?([-#+ 0,\(<]*)(\d+)?(?:\.(\d+))?[%A-Za-z]""", "g")
 
+  private def strOfZeros(count: Int): String = {
+    val twentyZeros = "00000000000000000000"
+    if (count <= 20) {
+      twentyZeros.substring(0, count)
+    } else {
+      var result = ""
+      var remaining = count
+      while (remaining > 20) {
+        result += twentyZeros
+        remaining -= 20
+      }
+      result + twentyZeros.substring(0, remaining)
+    }
+  }
+
+  @inline
+  private def assert(condition: Boolean, msg: => String): Unit = {
+    if (!condition)
+      throw new AssertionError(msg)
+  }
+
```

Comment:
For now this is fine, but I think it should go into some util, just like for javalanglib.